api/src/main/java/io/strimzi/api/kafka/CertificateExpirationPolicy.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public enum CertificateExpirationPolicy {
```

Comment:
This name is confusing me or I am missing something about its meaning ...

A CA certificate can be renewed for the following reasons:
* forced renew (using the annotation we already have)
* expires (it's in the expiring period)
* the related CA key is replaced

In the first two scenarios we are using the same CA key and they can be grouped under what you called `RENEW_CERTIFICATE` while the third scenario is related to the `REPLACE_KEY`.
Using the term "expiration" in the enum name let me think that it's about expiration/validity time while as far as I understood, these enum values define "the reason why a CA cert is renewed/regenerated", is that right?
Isn't it more a `CertificateRenewalPolicy" with `CERTIFICATE_EXPIRATION` and `REPLACE_KEY` ? (even if in the first enum value we will consider the forced renewal as well)

Reply:
This is enumerating the possibilities for what to do when the certificate is going to expire. We either generate a new certificate keeping the same key (what I've tried to consistently refer to as "certificate renewal"), or we generate a new key (and thus a new certificate too and is that I've tried to consistently call "key replacement"). 

If you want a certificate to be renewed (keeping the same key) you can use the `strimzi.io/force-renew` annotation on the relevant CA cert Secret.

Likewise if you want a key to be replaced (and thus replacing the certificate too) and can use the `strimzi.io/force-replace` annotation in the relevant CA key Secret. I decided to use different annotations because it reinforces the different terminology and means that if the user accidentally annotates the wrong `Secret` they will have done so with the wrong annotation, which will be harmless. Otherwise the risk is they accidentally annotate the key `Secret` and trigger key replacement, which would break the trust relationship for all their client apps. It's not unrecoverable (e.g. redeploy all the client apps), but it is inconvenient.

Reply:
> This is enumerating the possibilities for what to do when the certificate is going to expire.

Well, you could do a "key replacement" even when the certificate is not expiring but your key is compromised. In that case expiration isn't come to play, this is the reason why I think the enum name is misleading.

Reply:
But in that case you would annotate the key `Secret` with `strimzi.io/force-renew` and the property in the `Kafka` resource (i.e. this `enum`) is irrelevant. IOW you get the same effect as the two enumerated possibilities by using the (correct) annotations on the (correct) `Secrets`.